### DISCUSSION TITLE
PHP max file upload breaks the file upload Javascript

### DISCUSSION THREAD

**Discussion by joshhanley** [livewire/livewire#3026] [7. Bugs (archived)]
PHP has a default max file upload of 20

![image](https://user-images.githubusercontent.com/882837/116759251-3d297000-aa55-11eb-9a98-c5ded634d007.png)

But if a user uploads 21 files, rather than gracefully failing, this error is thrown

![image](https://user-images.githubusercontent.com/882837/116759290-53373080-aa55-11eb-9c09-f89ed9c9bc4b.png)

The issue is, if you view the request, PHP prepends the JSON output with HTML, causing the `JSON.parse` here to fail

https://github.com/livewire/livewire/blob/a909b1b9b0f31d28f569c4b22efebbbc17a4b81c/js/component/UploadManager.js#L115

This is what the request looks like 

![image](https://user-images.githubusercontent.com/882837/116759393-909bbe00-aa55-11eb-9b13-e2e57beba34b.png)

I think we can come up with a better way to handle that, probably by catching the error on the php end.

**Comment by calebporzio** [comment]
Agreed

**Comment by midblep** [comment]
Are there any plans to solve this? I kinda need it.

**Reply by joshhanley** [reply]
@Midblep I haven't the time at the moment, but if you need this, please submit a PR that fixes this, I'm sure it would be welcome 🙂

**Comment by lucasjose501** [comment]
What if, instead of sending all files in one request, livewire makes 1 request for each file?
